docs/advanced/guide/lifecycle.md

```diff
@@ -0,0 +1,3 @@
+# Test Lifecycle
+
+<!-- TODO: lifecyle diagram and reporter API -->
```

Comment:
Could we use `vitepress-plugin-mermaid` for this? 🤔 

https://emersonbottero.github.io/vitepress-plugin-mermaid/guide/styles.html

Reply:
Maybe when the lifecycle is finalised